src/main/scala/org/broadinstitute/hail/driver/FilterAlleles.scala

```diff
@@ -0,0 +1,156 @@
+
+package org.broadinstitute.hail.driver
+
+import scala.math.min
+import org.apache.spark.rdd.RDD
+import org.broadinstitute.hail.Utils._
+import org.broadinstitute.hail.annotations._
+import org.broadinstitute.hail.expr.{EvalContext, Parser, TArray, TBoolean, TGenotype, TInt, TSample, TVariant}
+import org.broadinstitute.hail.methods.Filter
+import org.broadinstitute.hail.variant.{AltAllele, GTPair, Genotype, Variant}
+import org.kohsuke.args4j.{Option => Args4jOption}
+
+import scala.collection.{GenTraversableOnce, mutable}
+import scala.reflect.ClassTag
+
+object FilterAlleles extends Command {
+  class Options extends BaseOptions {
+    @Args4jOption(required = false, name = "--keep", usage = "Keep variants matching condition")
+    var keep: Boolean = false
+
+    @Args4jOption(required = false, name = "--remove", usage = "Remove variants matching condition")
+    var remove: Boolean = false
+
+    @Args4jOption(required = true, name = "-c", aliases = Array("--condition"),
+      usage = "Filter expression involving <code>v</code> (variant), <code>va</code> (variant annotations), and <code>aIndex</code> (allele index)",
+      metaVar = "COND_EXPR")
+    var condition: String = _
+
+    @Args4jOption(required = true, name = "-a", aliases = Array("--annotation"),
+      usage = "Annotation modifying expression involving <code>v</code> (new variant), <code>va</code> (old variant annotations), and <code>aIndices</code> (maps from new to old indices)",
+      metaVar = "ANNO_EXPR")
+    var annotation: String = _
+  }
+
+  override def newOptions: Options = new Options
+
+  override def name: String = "filteralleles"
+
+  override def description: String = "Filter alleles in current dataset using the Hail expression language"
+
+  override def supportsMultiallelic: Boolean = true
+
+  override def requiresVDS: Boolean = true
+
+  override protected def run(state: State, options: Options): State = {
```

Comment:
I think you should check `vds.wasSplit` and if that's true, print a warning that this module is designed to handle multiallelics and that the user can use `filtervariants`